CONTRIBUTING.md

```diff
@@ -0,0 +1,74 @@
+## Contributing Guidelines
+
+### General
+
+* Fork the repo to your profile.
+* Make any changes in your forked repository.
+* All changes should be made in the 'main' branch. Do not create a new branch.
+* Make a pull request to this main repository.
+
+### Sample Projects
+
+Add all sample projects in the path 
+`/Samples/<your-sample-project-directory>/<your-c-file>.c`.
+
+There may be multiple users with their own sample file for the 
+sample projects. Do not edit the files contributed by other users.
+If naming conflict arises, append your username as a part of your 
+filename.
+
+#### List of sample projects you can contribute
+
+- Cellular Automaton 
+- 2D Particle Systems
+- Basic Interpreter (calculator with memory)
+- Fractals
+- Tetris
+- Basic HTTP Server
+- Compression + Decompression of text and binary files using Huffman Encoding (or other compression algorithm of choice)
+
+#### Sample Project Guidelines
+
+* Make use of small elegant libraries that are cross-platform.
+* We recommend a single source file (and a few if absolutely necessary).
+* Be sure to put instructions about downloading and installing the required libraries, link to relevant sites and you user names at the top of your source file as comments. An example format -
+
+```cpp
+/*
+ * Libraries
+ * SDL2 (link) - zlib license
+ * other libraries
+*/
+
+// Authors: name (github handle), name (github handle)
+
+/*
+ * Project Name
+small desc about the project and wiki links and such
+*/
+```
+
+#### Build Instructions
+
+These are general guidelines you can follow to keep the build 
+process consistent across all systems.
+
+Create build.bat (Windows) and build.sh (Linux) that contains the 
+build instructions for your program depending upon the system. The
+build file should be placed in the path - 
+`./Samples/<your-sample-project-directory>/bin/<build-script>`
```

Comment:
`./Samples/<your-sample-project-directory>/`
just there, the directory may look like:
main.c   build.bat   build.sh [bin]